.github/workflows/dev.yaml

```diff
@@ -3,16 +3,78 @@ name: build push dev image
 on:
   push:
     branches:
-      - 'main'
+      - 'issue1'
+  create:
+    tag:
 
 jobs:
+  testing:
+    runs-on: ubuntu-latest
+    container:
+      image: gsscogs/pythonversiontesting
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install poetry
+      run: |
+        poetry install
+        export python_dir=$(poetry run python -c "import site; print(site.getsitepackages()[0])") && patch -Nf -d "$python_dir/behave/formatter" -p1 < /cucumber-format.patch
+
+    - name: Run pyright
+      run: poetry run pyright . --lib
+
+    - name: Run unittests
+      if: always()
```

Comment:
Do we need `if: always()`? Will it always run if we remove it?

Reply:
we put it under every step to be safe, since the behave tests end up getting skipped when ever a unittests failed